### PR TITLE
PCA-dashboard

### PR DESCRIPTION
**Pull request by LadyBluenotes**
Using this to check to see how it runs or if it can possibly pass tests

### CODE REVIEW

**Comment by LadyBluenotes** [comment]
Hey @Sulstice - Wondering if you'd be able to give a quick rundown on the checks? I think knowing what is passing or failing and why can help me understand how to fix them.

**Comment by Sulstice** [comment]
NextJS is a new framework for me around React, I think I would need to have you demo it to me first. I see you put the infrastructure there and it's pretty intuitive so far. I removed the SVG files you committed. We don't need to pollute the codebase with stuff we won't be using.

For the Jobs, this is definitely my fault going crazy with bots. 

- Pre-commit is a static analysis for code quality, the pre-commit is focused on python standards which means it would need to change for CSS or JS. Perhaps you could modify that. That is usually to ensure consistency on a larger scale. 
- GlobalChemAPI - Is the API for testing the back-end code. That should always be stable unless something drastic changes. Look out for that one. The GlobalChem API is failiing on windows which has been an issue for me in the past so I need to go look that up and one of the previous issues with that. 
- GlobalChemExtensions API - should be cut as a job to run on commits for PRs. If you want I can show you how to regulate Github actions. 
- Mergify - Kicks in when rules have been met on a PR which allow it for automatic moving into the main code base. Usually if all tests past and a reviewer has gone in an reviewed the work.

**Comment by LadyBluenotes** [comment]
We can always chat more about it when we meet up tomorrow, but if you're curious (or anyone else is) about why it's a nice framework, I really like that they have server-side rendering, automatic code splitting, and pre-fetching of data. They also have some static site generation, built-in API route system, and image optimization.

While a lot of these can be beneficial for larger apps, if we ever care to build on top of things for one reason or another Next can make it easier. With that being said, we could always just keep to React, if you feel like that's more appropriate.

**Comment by Sulstice** [comment]
@LadyBluenotes I added the code for a github workflow. It outputs a html file. I think we can use github actions to run a "Job" What do you think?

File: .github/workflows/discord_pca_analysis.yml
Comment (ANUGAMAGE) [inline .github/workflows/discord_pca_analysis.yml:1]: Sul and Sara,
I am hoping to merge this Pr to the main branch. If you have time please review before merging.

**Review by ANUGAMAGE — COMMENTED** [review]
going to merge the code

**Pull request merged**